### ISSUE TITLE
start_link/1 with all options

### ISSUE THREAD

**Issue by LostKobrakai** [lucaong/cubdb#9]
The child_spec supervisor notation of `{CubDB, opts}` normalized to calling `start_link(opts)`. CubDB though receives additional options as additional parameters.

**Comment by lucaong** [comment]
Hi @LostKobrakai ! Sorry, somehow I missed your issue.

You are 100% right. I am in fact already working on improving this. The improvement comes in two parts:

  1. I changed already the `child_spec` function in the latest release, to allow passing all options when starting `CubDB` as part of a supervision tree. This makes it possible to pass this in the supervisor `children` list: `{CubDB, data_dir: "somedir", auto_compact: true, name: MyApp.DB}` (note that I am passing both CubDB options and genserver options like `name`)

  2. I am now working on providing a version of `start_link` taking a single argument. I am trying to balance the need for a clean API with the wish to make this change as backward compatible as possible.

**Comment by LostKobrakai** [comment]
I mean in the end the `child_spec/1` alone is actually enough for what I asked for, but I guess a `start_link/1` will prevent surprises if someone just copies args from a supervisor.

**Comment by lucaong** [comment]
Yes, I think that having a `start_link/1` is better, and will make the custom `child_spec/1` unnecessary.

I want to maintain partial backward compatibility, so what I am working on is the following:

```elixir
# Passing only the `data_dir` as a string is still allowed:
{:ok, pid} = CubDB.start_link("some/data/dir")

# Passing a single keyword argument with options:
#  - data_dir option is required, this will return {:error, cause} if not given
#  - GenServer options like `name` or `timeout` are passed to GenServer.start_link/3
{:ok, pid} = CubDB.start_link(data_dir: "some/data/dir", auto_compact: true, name: :db)

# Passing two arguments (data_dir and options) is still allowed:
{:ok, pid} = CubDB.start_link("some/data/dir", auto_compact: true, name: :db)
```

Passing 3 arguments (data_dir, options, gen_server_option) would not be allowed anymore.
This is a small breaking change, but maintaining this would make the API more complex for little gain. The 1 or 2 args versions can do all that the 3 args version was doing anyway.

The other small breaking change would be that passing the data dir as a charlist would not be supported anymore (one would have to call `List.to_string` before). This is because distinguishing between a charlist or a keyword list when passing a single argument would need some logic that I would prefer to avoid entirely. I'd prefer to introduce some breaking change before `v1.0` than to support lots of variants of the same function.

Thoughts on this?

**Comment by LostKobrakai** [comment]
I'm fine with that.

**Comment by lucaong** [comment]
Here is the PR for this change: https://github.com/lucaong/cubdb/pull/11

It will be part of the upcoming release, together with https://github.com/lucaong/cubdb/pull/10

**Comment by lucaong** [comment]
I released `v0.16.0` including this change.
